### DISCUSSION TITLE
How to user Kubernetes to send a HTTP request?

### DISCUSSION THREAD

**Discussion by zxyao145** [kubernetes-client/csharp#1366] [Q&A]
I am trying to access the RESTful API of K8s, but why is kubernetes.HttpClient.BaseAddress null? And how should I send an HTTP request? 

Looking forward to receiving an answer, thank you

![image](https://github.com/kubernetes-client/csharp/assets/24973084/005f6bba-68e0-4fcd-8178-d2143291fb9b)

**Comment by tg123** [comment]
did not set the prop for httpclient
you can use full uri to send request to api server

**Reply by tg123** [reply]
btw client has baseuri prop, seems that is what you are looking for

**Comment by zxyao145** [comment]
> btw client has baseuri prop, seems that is what you are looking for顺便说一句，客户端有baseuri prop，似乎这就是你正在寻找的

Yes,  it is. But why do developers need to manually set it up instead of encapsulating it internally in the SDK?
